src/views/Timesheets/index.js

```diff
@@ -0,0 +1,539 @@
+      let timeDiff = timeOut.getTime() - timeIn.getTime();
+      let calculatedTimeDiff = timeDiff / 1000 / 60 / 60;
+      let roundedHourDifference = (Math.round(calculatedTimeDiff * 4) / 4).toFixed(2);
+      setHoursWorkedInDecimal(roundedHourDifference);
+      let hoursWorked = Math.floor(calculatedTimeDiff);
+      let minutesWorked = Math.round((calculatedTimeDiff - hoursWorked) * 60);
+
+      if (minutesWorked >= 60) {
+        hoursWorked++;
+        minutesWorked = 0;
+      }
+
+      if (timeDiff < 0) {
+        setTimeOutIsBeforeTimeIn(true);
+      } else {
+        setTimeOutIsBeforeTimeIn(false);
+
+      }
+      if (calculatedTimeDiff > 20) {
+        setShiftTooLong(true);
+      } else {
+        setShiftTooLong(false);
+      }
+
+      if (calculatedTimeDiff < 0.25) {
+        setShiftTooShort(true);
+      } else {
+        setShiftTooShort(false);
+      }
+    }
+  };
+
+  const checkForFutureDate = () => {
+    let now = Date.now();
+    setEnteredFutureTime((selectedDateIn.getTime() > now) || (selectedDateOut.getTime() > now));
+  }
+
+  if (props.Authentication) {
+    function Alert(props) {
+      return <MuiAlert elevation={6} variant="filled" {...props} />;
+    }
+
+    const getActiveJobsForUser = () => {
+      let details = {
+        shift_start_datetime: selectedDateIn.toLocaleString(),
+        shift_end_datetime: selectedDateOut.toLocaleString(),
+      };
+      jobs.getActiveJobsForUser(details).then(result => {
+        setUserJobs(result);
+      });
+    };
+
+    const getSavedShiftsForUser = () => {
+      return jobs.getSavedShiftsForUser();
+    };
+
+    const handleDateChange1 = date => {
+      setSelectedDateIn(date);
+      handleTimeOutIsBeforeTimeIn(date, selectedDateOut);
+    };
+
+    const handleDateChange2 = date => {
+      setSelectedDateOut(date);
+      handleTimeOutIsBeforeTimeIn(selectedDateIn, date);
+    };
+
+    const handleSaveButtonClick = () => {
+      let timeIn = selectedDateIn;
+      let timeOut = selectedDateOut;
+
+      if (selectedDateIn.getDay() === 6 && selectedDateOut.getDay() === 0) {
+        let timeOut2 = new Date(timeOut.getTime());
+        let timeIn2 = new Date(timeOut.getTime());
+        timeIn2.setHours(0);
+        timeIn2.setMinutes(0);
+        timeIn2.setSeconds(0);
+        timeIn2.setMilliseconds(0);
+
+        timeOut.setDate(timeIn.getDate());
+        timeOut.setHours(23);
+        timeOut.setMinutes(59);
+        timeOut.setSeconds(0);
+        timeOut.setMilliseconds(0);
+
+        let timeDiff2 = timeOut2.getTime() - timeIn2.getTime();
+        let calculatedTimeDiff2 = timeDiff2 / 1000 / 60 / 60;
+        let roundedHourDifference2 = (Math.round(calculatedTimeDiff2 * 4) / 4).toFixed(2);
+
+        saveShift(
+          selectedJob.EMLID,
+          timeIn2.toLocaleString(),
+          timeOut2.toLocaleString(),
+          roundedHourDifference2,
+          userShiftNotes,
+        ).then(result => {
+          setSnackbarOpen(true);
```

Comment:
Do you need to do the same cleanup here as you did at 152? (setting things to null and 0, etc.)

Reply:
It will reach that code that sets them null every time this method is run

Reply:
Oh, okay!